src/tls.rs

```diff
@@ -667,6 +792,18 @@ mod esptls {
             res
         }
 
+        /// Establish a TLS/SSL connection using the adopted connection, acting as the server.
+        ///
+        /// # Errors
+        ///
+        /// * `ESP_FAIL` if connection could not be established
+        #[cfg(esp_idf_esp_tls_server)]
+        pub async fn negotiate_server(&mut self, cfg: &ServerConfig<'_>) -> Result<(), EspError> {
+            // FIXME: this isn't actually async, but esp-idf does not expose anything else.
```

Comment:
I don't see a good solution.

ESP IDF's `esp_mbedtls_server_session_create` seems simply not created with `async` in mind. Fixing it is not rocket science but would require ESP IDF upstream to accept a PR, where `esp_mbedtls_server_session_create` (and likely its WolfSSL counterpart) is re-written by first splitting it in two:
- `esp_mbedtls_server_session_create_start`
-  `esp_mbedtls_server_session_create_continue_async`

... and then re-assembling it by calling the above two new methods.

... where the current `while` loop is moved into the `_async` method but not as a loop, but as a simple "if" (because the looping will be in the user code, inter-mingled with `.await`-ing the socket for data to become available/write buffer space to become available).

Reply:
I agree. I might do a PR for upstream but at the moment it's not a priority for me.

Reply:
Would you mind the commenting out / removing the `negotiate_server` method from the async TLS? In its current shape it is not usable, so we should rather not have it in the first place.